modin/pandas/groupby.py

```diff
@@ -651,8 +651,11 @@ def __getitem__(self, key):
                         + "df.groupby(df['by_column'].copy())['by_column']"
                     ),
                 )
-            cols_to_grab = internal_by.union(key)
-            key = [col for col in self._df.columns if col in cols_to_grab]
+            # We need to maintain order of the columns in key, using a set doesn't
+            # maintain order
+            cols_to_grab = list(self._internal_by)
+            cols_to_grab.extend(x for x in key if x not in cols_to_grab)
```

Comment:
this is having an `O(N*M)` complexity... I think that _maybe_ using a dictionary with some dummy values could be faster, as dictionaries maintain insertion order since 3.7 AFAIR, like
```python
cols_to_grab = dict.fromkeys(self._internal_by)
cols_to_grab.update(dict.fromkeys(key))
```

Reply:
My concern here is that this may complicate the code a bit, but I've applied the change, and added a comment, so hopefully it should be ok!